sql/core/src/test/scala/org/apache/spark/sql/streaming/FileStreamSourceSuite.scala

```diff
@@ -1153,174 +1153,213 @@ class FileStreamSourceSuite extends FileStreamSourceTest {
+              val q = df.writeStream.format("memory").queryName(testTable).start()
+              try {
+                q.processAllAvailable()
+              } finally {
+                q.stop()
+              }
+            }
+            assert(e.getCause.isInstanceOf[IllegalArgumentException])
+            Seq(optionName, value, "positive integer").foreach { s =>
+              assert(e.getMessage.contains(s))
             }
           }
-          assert(e.getCause.isInstanceOf[IllegalArgumentException])
-          Seq("maxFilesPerTrigger", value, "positive integer").foreach { s =>
-            assert(e.getMessage.contains(s))
-          }
-        }
 
-        testMaxFilePerTriggerValue("not-a-integer")
-        testMaxFilePerTriggerValue("-1")
-        testMaxFilePerTriggerValue("0")
-        testMaxFilePerTriggerValue("10.1")
+          testMaxFilePerTriggerValue("not-a-integer")
+          testMaxFilePerTriggerValue("-1")
+          testMaxFilePerTriggerValue("0")
+          testMaxFilePerTriggerValue("10.1")
+        }
       }
     }
   }
 
-  test("SPARK-30669: maxFilesPerTrigger - ignored when using Trigger.Once") {
-    withTempDirs { (src, target) =>
-      val checkpoint = new File(target, "chk").getCanonicalPath
-      val targetDir = new File(target, "data").getCanonicalPath
-      var lastFileModTime: Option[Long] = None
-
-      /** Create a text file with a single data item */
-      def createFile(data: Int): File = {
-        val file = stringToFile(new File(src, s"$data.txt"), data.toString)
-        if (lastFileModTime.nonEmpty) file.setLastModified(lastFileModTime.get + 1000)
-        lastFileModTime = Some(file.lastModified)
-        file
+  testQuietly("max bytes per trigger & max files per trigger - both set") {
+    val testTable = "maxBytesPerTrigger_maxFilesPerTrigger_test"
+    withTable(testTable) {
+      withTempDir { case src =>
+        val df = spark.readStream
+          .option("maxBytesPerTrigger", "1")
+          .option("maxFilesPerTrigger", "1")
+          .text(src.getCanonicalPath)
+        val e = intercept[StreamingQueryException] {
+          // Note: a tested option is checked in the stream thread when creating the source
+          val q = df.writeStream.format("memory").queryName(testTable).start()
+          try {
+            q.processAllAvailable()
+          } finally {
+            q.stop()
+          }
+        }
+        assert(e.getCause.isInstanceOf[IllegalArgumentException])
+        Seq("maxBytesPerTrigger", "maxFilesPerTrigger", "can't be both set").foreach { s =>
+          assert(e.getMessage.contains(s))
+        }
       }
+    }
+  }
 
-      createFile(1)
-      createFile(2)
-      createFile(3)
-
-      // Set up a query to read text files one at a time
-      val df = spark
-        .readStream
-        .option("maxFilesPerTrigger", 1)
-        .text(src.getCanonicalPath)
+  test("SPARK-30669: maxFilesPerTrigger & maxBytesPerTrigger - ignored when using Trigger.Once") {
```

Comment:
ditto. Please avoid touching the existing SPARK-30669 test case .

Reply:
fixed